rayon-core/src/configuration.rs

```diff
@@ -77,7 +58,7 @@ impl Configuration {
 
     /// Get the number of threads that will be used for the thread
     /// pool. See `set_num_threads` for more information.
-    pub fn num_threads(&self) -> Option<usize> {
+    pub fn num_threads(&self) -> usize {
         self.num_threads
```

Comment:
I think we should make this reflect the actual number that will be used.  So we should move the logic with `RAYON_RS_NUM_CPUS` and `num_cpus::get()` fallbacks here.